tests/v2/validation/crdgeneration/crd_generation.go

```diff
@@ -0,0 +1,52 @@
+package crdgeneration
+
+import "strings"
+
+const (
+	groupManagement   = "management.cattle.io"
+	groupCatalog      = "catalog.cattle.io"
+	groupUI           = "ui.cattle.io"
+	groupCluster      = "cluster.cattle.io"
+	groupProject      = "project.cattle.io"
+	groupRKE          = "rke.cattle.io"
+	groupProvisioning = "provisioning.cattle.io"
+	groupFleet        = "fleet.cattle.io"
+	roleTemplate      = "roletemplates.management.cattle.io" //Remove in 2.8.0
+)
+
+func mapCRD(crdList []string) map[string][]string {
+	crdMap := map[string][]string{
+		groupManagement:   {},
+		groupCatalog:      {},
+		groupUI:           {},
+		groupCluster:      {},
+		groupProject:      {},
+		groupRKE:          {},
+		groupProvisioning: {},
+		groupFleet:        {},
+	}
+
+	for _, crd := range crdList {
+		if strings.Contains(crd, groupManagement) {
+			crdMap[groupManagement] = append(crdMap[groupManagement], crd)
+		} else if strings.Contains(crd, groupCatalog) {
+			crdMap[groupCatalog] = append(crdMap[groupCatalog], crd)
+		} else if strings.Contains(crd, groupUI) {
+			crdMap[groupUI] = append(crdMap[groupUI], crd)
+		} else if strings.Contains(crd, groupCluster) {
+			crdMap[groupCluster] = append(crdMap[groupCluster], crd)
+		} else if strings.Contains(crd, groupProject) {
+			crdMap[groupProject] = append(crdMap[groupProject], crd)
+		} else if strings.Contains(crd, groupRKE) {
+			crdMap[groupRKE] = append(crdMap[groupRKE], crd)
+		} else if strings.Contains(crd, groupProvisioning) {
+			crdMap[groupProvisioning] = append(crdMap[groupProvisioning], crd)
+		} else if strings.Contains(crd, groupFleet) {
+			crdMap[groupFleet] = append(crdMap[groupFleet], crd)
+		} else {
+			crdMap["additional"] = append(crdMap["additional"], crd)
+		}
+	}
+
+	return crdMap
+}
```

Comment:
Just an idea for this section for less `else if` statements. Also, if a group ever needed to be added or removed it could just be added to or removed from the slice. Just an optional idea

```go
func mapCRD(crdList []string) map[string][]string {
	groups := []string{groupManagement, groupCatalog, groupUI, groupCluster, groupProject, groupRKE, groupProvisioning, groupFleet}
	crdMap := make(map[string][]string)

	for _, group := range groups {
		crdMap[group] = []string{}
	}

	crdMap["additional"] = []string{}

	for _, crd := range crdList {
		found := false
		for _, group := range groups {
			if strings.Contains(crd, group) {
				crdMap[group] = append(crdMap[group], crd)
				found = true
				break
			}
		}
		if !found {
			crdMap["additional"] = append(crdMap["additional"], crd)
		}
	}

	return crdMap
}

```

Reply:
yeah I think both has its pros and cons. using multiple if else might give us better readability but as you rightly pointed it adds complexity if we ever need to add/remove a group and more code duplication. Updated the PR to include the suggestion.